### PR TITLE
style: convention for optional reference wrapper type alias

### PR DESCRIPTION
**Pull request by junr03**
Description: noticed in #9516 that we did not have a convention around aliasing `absl::optional<std::reference_wrapper<T>>`. This PR proposes one.
Risk Level: low. compiles.
Testing: compilation and test suite ran.
Docs Changes: added the convention in STYLE.md

Signed-off-by: Jose Nino <jnino@lyft.com>

### CODE REVIEW

**Comment by junr03** [comment]
@jmarantz this is kind of related to #9710. I was chatting with @derekargueta about placing format checking for smart pointer and for this type of aliases. I can create an issue to track if you think it is worth to enforce.

File: STYLE.md [hunk 1]
```diff
@@ -43,6 +43,9 @@
   * `using BarSharedPtr = std::shared_ptr<Bar>;`
   * `using BlahConstSharedPtr = std::shared_ptr<const Blah>;`
   * Regular pointers (e.g. `int* foo`) should not be type aliased.
+* `absl::optional<std::reference_wrapper<T>> is type aliased:
+  * `using FooOptRef = absl::optional<std::reference_wrapper<T>>;`
+  * `using FooOptConstRef = absl::optional<std::reference_wrapper<T>>;`
```
Comment (derekargueta) [inline STYLE.md]: We should probably expand the language on line 41 beyond "smart pointers".

Comment (junr03) [reply STYLE.md]: @derekargueta not sure what you mean? I see these two as different bullet points.

Comment (derekargueta) [reply STYLE.md]: oh sorry, missed the un-indentation on 46 to make it a new section 👍

File: include/envoy/stats/scope.h [hunk 2]
```diff
@@ -19,9 +19,9 @@ class Histogram;
 class Scope;
 class NullGaugeImpl;
 
-using OptionalCounter = absl::optional<std::reference_wrapper<const Counter>>;
-using OptionalGauge = absl::optional<std::reference_wrapper<const Gauge>>;
-using OptionalHistogram = absl::optional<std::reference_wrapper<const Histogram>>;
+using CounterOptRef = absl::optional<std::reference_wrapper<const Counter>>;
```
Comment (jmarantz) [inline include/envoy/stats/scope.h]: should this be CounterOptConstRef per your style section?

Comment (junr03) [reply include/envoy/stats/scope.h]: Yeah, I should definitely follow my rules 😅. I checked all the other aliases for correctness.

Comment (junr03) [reply include/envoy/stats/scope.h]: Overall I would like the aliases to be CI checked in our formatting run. I will open an issue for that if someone wants to pick up

Comment (junr03) [reply include/envoy/stats/scope.h]: https://github.com/envoyproxy/envoy/issues/9840

**Pull request merged**